### PR TITLE
[HER-89] Configure fetching in camera repository

### PR DESCRIPTION
**Pull request by Urzica-Platon-Casian**
## Description
-    it will live inside the Data layer
-    it will receive a PCamera
-    it will return an IEnumerable<MCamera>
-    it will be used by the business command

## Related
[HER-89](https://nerds-interns.atlassian.net/jira/software/projects/HER/boards/1?issueParent=10022&selectedIssue=HER-89)

### CODE REVIEW

File: src/Data/Camera/CameraRepository.cs [hunk 1]
```diff
@@ -21,6 +22,10 @@ public CameraRepository(Context context)
             .Select(CameraFactory.MakeModel)
             .ToList();
 
+        public IEnumerable<MCamera> ByParameter(PCamera parameter) => _context.Set<ECamera>()
+            .ToList()
+            .Select(CameraFactory.MakeModel);
```
Comment (RaduBuciuceanu) [inline src/Data/Camera/CameraRepository.cs:27]: Should we also use our parameter to filter the output here?

Comment (RaduBuciuceanu) [reply src/Data/Camera/CameraRepository.cs:27]: @danidoff Can you help us here, please?

**Review by RaduBuciuceanu — CHANGES_REQUESTED** [review]

**Review by TBiannca — DISMISSED** [review]

File: src/Data/Camera/CameraRepository.cs (hunk 1 above)
Comment (Urzica-Platon-Casian) [reply src/Data/Camera/CameraRepository.cs]: Done

File: src/Data/Camera/Filtering/CameraFilter.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    using Business.Camera.Fetching.Models;
+
+    public class CameraFilter : ICameraFilter
+    {
+        private PCamera _parameter;
+
+        public IQueryable<ECamera> Execute(IQueryable<ECamera> input) =>
+            input.Where(entity => _parameter.Ids.Contains(entity.Id));
+
+        public ICameraFilter With(PCamera parameter)
```
Comment (danidoff) [inline src/Data/Camera/Filtering/CameraFilter.cs:13]: Should we move this method above the Execute ?

File: src/Data/Camera/Filtering/CameraFilter.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    using System.Linq;
+    using Business.Camera.Fetching.Models;
+
+    public class CameraFilter : ICameraFilter
+    {
+        private PCamera _parameter;
+
+        public IQueryable<ECamera> Execute(IQueryable<ECamera> input) =>
+            input.Where(entity => _parameter.Ids.Contains(entity.Id));
```
Comment (danidoff) [inline src/Data/Camera/Filtering/CameraFilter.cs:11]: Should we check if parameter.Ids is not null here ?

**Review by danidoff — CHANGES_REQUESTED** [review]

File: src/Data/Camera/Filtering/ICameraFilter.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+namespace Data.Camera.Filtering
+{
+    using System.Linq;
+    using Business;
+    using Business.Camera.Fetching.Models;
+
+    public interface ICameraFilter : ICommand<IQueryable<ECamera>, IQueryable<ECamera>>
```
Comment (danidoff) [inline src/Data/Camera/Filtering/ICameraFilter.cs:7]: Should we register this Interface on IoC ?

**Review by danidoff — CHANGES_REQUESTED** [review]

File: src/Data/Camera/Filtering/CameraFilter.cs (hunk 2 above)
Comment (Urzica-Platon-Casian) [reply src/Data/Camera/Filtering/CameraFilter.cs:10]: Done

File: src/Data/Camera/Filtering/CameraFilter.cs (hunk 3 above)
Comment (Urzica-Platon-Casian) [reply src/Data/Camera/Filtering/CameraFilter.cs]: Done

File: src/Data/Camera/Filtering/ICameraFilter.cs (hunk 4 above)
Comment (Urzica-Platon-Casian) [reply src/Data/Camera/Filtering/ICameraFilter.cs:7]: Done

**Review by danidoff — APPROVED** [review]

**Review by danidoff — APPROVED** [review]

**Review by RaduBuciuceanu — APPROVED** [review]

**Pull request merged**